python/pyarrow/tests/interchange/test_extra.py

```diff
@@ -0,0 +1,371 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Maybe call this file something like `test_conversion.py`? (more descriptive than `test_extra.py`, and I think this file is mostly checking the actual conversion, and the other file is more testing the API of the spec object?)